.travis.yml

```diff
@@ -20,11 +18,17 @@ matrix:
     - env: DOCKER="ubuntu-trusty-x86"
     - env: DOCKER="ubuntu-xenial-amd64"
     - env: DOCKER="debian-stretch-x86"
-
 install:
-    - sudo apt-get update && sudo apt-get install -qyy debootstrap
-    - sudo chown -R 1000 $TRAVIS_BUILD_DIR
-    - cd "$DOCKER" && make update
-
+- sudo apt-get update && sudo apt-get install -qyy debootstrap
+- sudo chown -R 1000 $TRAVIS_BUILD_DIR
+- cd "$DOCKER" && make update
 script:
-    - make build && make test
+- if [[ -n "$DOCKER_USERNAME" ]]; then docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+  ; fi
+- make build && make test
+after_success:
+- if [[ "$TRAVIS_BRANCH" == "master" ]]; then make push ; fi
```

Comment:
$TRAVIS_BRANCH is ambiguous depending on whether this is a PR or not, see:

https://graysonkoonce.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/

This attempts an upload for PRs too. It needs an extra against $TRAVIS_PULL_REQUEST.